autonomic-plugin-common/src/main/java/br/com/autonomiccs/autonomic/plugin/common/guru/AutonomiccsSystemVmsGuru.java

```diff
@@ -142,49 +169,66 @@ public void finalizeStop(VirtualMachineProfile arg0, Answer arg1) {
     }
 
+    /**
+     * This method orchestrates the virtual machine profile configuration.
+     */
     @Override
     public boolean finalizeVirtualMachineProfile(VirtualMachineProfile profile, DeployDestination dest, ReservationContext context) {
-        StringBuilder buf = profile.getBootArgsBuilder();
-        buf.append(" template=domP");
-        buf.append(" host=").append(ApiServiceConfiguration.ManagementHostIPAdr.value());
-        buf.append(" name=").append(profile.getVirtualMachine().getHostName());
-        buf.append(" zone=").append(dest.getDataCenter().getId());
-        buf.append(" pod=").append(dest.getPod().getId());
-        buf.append(" guid=autonomiccsSystemVm.").append(profile.getId());
-        buf.append(" autonomiccsSystemVm_vm=").append(profile.getId());
+        StringBuilder buf = configureBasicVmProfile(profile, dest);
 
         boolean externalDhcp = false;
         String externalDhcpStr = configurationDao.getValue("direct.attach.network.externalIpAllocator.enabled");
         if (externalDhcpStr != null && externalDhcpStr.equalsIgnoreCase("true")) {
             externalDhcp = true;
         }
-        if (Boolean.valueOf(configurationDao.getValue("system.vm.random.password"))) {
-            buf.append(" vmpassword=").append(configurationDao.getValue("system.vm.password"));
-        }
+        configurePassword(buf);
 
         for (NicProfile nic : profile.getNics()) {
             int deviceId = nic.getDeviceId();
-            if (nic.getIPv4Address() == null) {
-                buf.append(" eth").append(deviceId).append("ip=").append(ipAddressInternet);
-                buf.append(" eth").append(deviceId).append("mask=").append(maskForIpAddressInternet);
-            } else {
-                buf.append(" eth").append(deviceId).append("ip=").append(nic.getIPv4Address());
-                buf.append(" eth").append(deviceId).append("mask=").append(nic.getIPv4Netmask());
-            }
+            configureIpAndMaskOnNicWithIpAddressNullOrNotNull(buf, nic, deviceId);
+            configureNicGatewayAndManagementNetwork(dest, buf, nic);
+        }
 
-            if (nic.isDefaultNic()) {
-                buf.append(" gateway=").append(nic.getIPv4Gateway());
-            }
+        String bootArgs = configureNicDhcpAndDns(profile, buf, externalDhcp);
 
-            if (nic.getTrafficType() == TrafficType.Management) {
-                String mgmt_cidr = configurationDao.getValue(Config.ManagementNetwork.key());
-                if (NetUtils.isValidCIDR(mgmt_cidr)) {
-                    buf.append(" mgmtcidr=").append(mgmt_cidr);
-                }
-                buf.append(" localgw=").append(dest.getPod().getGateway());
-            }
+        logger.debug("Boot Args for " + profile + ": " + bootArgs);
+        return true;
+    }
+
+    /**
+     * It configures the password, the String has a ormat as the following example
```

Comment:
It configures the password, the String has the format as the following example "vmpassword=123456"